### PR TITLE
[Security] [LoginLink] Throw InvalidLoginLinkException on missing parameter

### LINKED ISSUE #48291: [Security][LoginLink] Request missing parameters throws 500
### Symfony version(s) affected

6.1.2

### Description

When login_check is requested and the request parameters are missing (user, hash, expires) it throws 500 exceptions as verifySignatureHash method needs 3 not null parameters .

### How to reproduce

Implement LoginLink and then request login via link:
`https://mydomain.eu/login_check?amp%3Bexpires=1669158801&amp%3Bhash=YTU2NzdiNjE3ZGZhZRTHJH5678c1ODcyOTAyNDc0MzhiFiZmRlOWRjMjE4NDhhOGE4ZDcxNjU5OTUzNw%3D%3D%7CYTg5Y2JlNjY4MmQxZWU0M...5lc3M&user=myemail%40example.com`

You see that here all parameters are provided but actually they are not. Due to some weird solution in our client's corporation when they get an email with the link, their inner security solution encodes ampersand to `amp%3B`. This makes those parameters obsolete. 

### Possible Solution

We should check in `vendor/symfony/security-http/LoginLink/LoginLinkHandler.php:92` if those parameters are not null and if yes, then throw an exception that can by handled.

### Additional Context

`Uncaught Error: Symfony\Component\Security\Core\Signature\SignatureHasher::verifySignatureHash(): Argument #2 ($expires) must be of type int, null given, called in /vendor/symfony/security-http/LoginLink/LoginLinkHandler.php on line 96`

### PR DESCRIPTION
**Pull request by MatTheCat**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #48291
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

**Review by alamirault — APPROVED** [review]

File: src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php [hunk 1]
```diff
@@ -97,8 +97,12 @@ public function consumeLoginLink(Request $request): UserInterface
             throw new InvalidLoginLinkException('User not found.', 0, $exception);
         }
 
-        $hash = $request->get('hash');
-        $expires = $request->get('expires');
+        if (!($hash = $request->get('hash'))) {
```
Comment (fabpot) [inline src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php]:
```suggestion
        if (!$hash = $request->get('hash')) {
```

File: src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php [hunk 2]
```diff
@@ -97,8 +97,12 @@ public function consumeLoginLink(Request $request): UserInterface
...
+            throw new InvalidLoginLinkException('Missing "hash" parameter.');
+        }
+        if (!($expires = $request->get('expires'))) {
```
Comment (fabpot) [inline src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php]:
```suggestion
        if (!$expires = $request->get('expires')) {
```

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php [hunk 3]
```diff
@@ -97,8 +97,12 @@ public function consumeLoginLink(Request $request): UserInterface
             throw new InvalidLoginLinkException('User not found.', 0, $exception);
         }
 
-        $hash = $request->get('hash');
-        $expires = $request->get('expires');
+        if (!$hash = $request->get('hash')) {
+            throw new InvalidLoginLinkException('Missing "hash" parameter.');
+        }
+        if (!$expires = $request->get('expires')) {
+            throw new InvalidLoginLinkException('Missing "expires" parameter.');
```
Comment (chalasr) [inline src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php:104]: Other authenticators throw a `BadRequestHttpException` in such cases, I would do the same here. See e.g. https://github.com/symfony/symfony/blob/c79d4ab7e787eb27d83581ccb8e15a4e11e88df3/src/Symfony/Component/Security/Http/Authenticator/JsonLoginAuthenticator.php#L161

**Review by chalasr — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php (hunk 3 above)
Comment (MatTheCat) [reply src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php:104]: Hmm `LoginLinkHandlerInterface::consumeLoginLink` is supposed to throw `InvalidLoginLinkException`: https://github.com/symfony/symfony/blob/f67ff7640624a53b3f0d096a2d7dbba5a5369001/src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandlerInterface.php#L32-L34

The `LoginLinkAuthenticator` then converts it to `InvalidLoginLinkAuthenticationException` which triggers its `onAuthenticationFailure` method to be called.

Are you sure we should bypass this behavior?

Comment (chalasr) [reply src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php:104]: Thinking twice about it, I think it's safer to go with your approach.

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @MatTheCat.

**Pull request merged**